### DISCUSSION TITLE
8.1.0 (Light SDK and LightFX update)

### DISCUSSION THREAD

**Discussion by T-Troll** [T-Troll/alienfx-tools#289] [Announcements]
### **Warning! Some older configuration formats not supported anymore! Backup your configuration using `alienfx-config`!**
- In case you upgrade from 7.7.2.1 or early, first install and run [7.8.1](https://github.com/T-Troll/alienfx-tools/releases/tag/7.8.1)
- In case you upgrade from 7.10.5 or early, first install and run [7.10.6.2](https://github.com/T-Troll/alienfx-tools/releases/tag/7.10.6.2)

###  Changes into this release:

`All light-control tools`:
- New SDK version with different interface - code is smaller and faster.

`AlienFX-GUI`:
- Bug fixed: Global effect can not reset in some cases.
- Bug fixed: Global effect not set if no lights changes in update.
- Bug fixed: Ghost icon in tray if start in non-default mode (dimmed, lights off).

`AlienFX-CLI`:
- Effect speed and length have default values now.
- Effect speed now in ms into `settempo` command.
- Bug fixed: Zero brightness for high-level API
- Bug fixed: Incorrect action type into `setaction` command for high-level API.
- Bug fixed: Red and Blue components swapped for some commands for High-level API.

`LightFX.dll (Dell API emulator)`
- New: all grids used to detect light position.
- New: NumDevs() method now re-detect devices at call (it's by API description, to detect device connection/disconnection).
- Performance increased about 50 times for APIv5 and v8 (RGB keyboards).
- Indicator lights now excluded from zone operations (as well as power light before).
- Bug fixed: Incorrect light set for LFX_ALL.

<hr /><em>This discussion was created from the release <a href='https://github.com/T-Troll/alienfx-tools/releases/tag/8.1.0'>8.1.0 (Light SDK and LightFX update)</a>.</em>

**Comment by MuaazH** [comment]
Can you please update the "how to use" in the SDK? It's still says use `SetColor`.
Tested on G7 7700 laptop, you can add it to the list, it's just like G5 5500. It works great. (But does not turn-off lights if you delete them).
Many thanks.

G7 7700 Lights = 3 Main keyboard + 1 Numpad + 12 on a straight line under the keyboard.

**Reply by T-Troll** [reply]
> Can you please update the "how to use" in the SDK? It's still says use SetColor.

SDK or LightFX emulator?
Yes, i keep SetColor into [SDK](https://github.com/T-Troll/AlienFX-SDK) for backward compatibility. But parameters changed in latest release for unification - all methods now use Afx_Action instead of Afx_colorcode.

It's not the final change, however, i plan to rebuild Afx_lightblock to keep more, then one lightID into one set - it will increase update performance for both tron and RGB keyboards dramatically (they both support multi-light set to one value in hardware).

> But does not turn-off lights if you delete them

In case you mean "delete light mapping" - it's ok. Removing mapping means "don't touch me anymore!". But it still participated in case device have global hardware brightness (APIv0, v5, v8 have it, v1-v3 have global off, but manual on, v4 and v6 have per-light control).

> Tested on G7 7700 laptop,

I have its mappings into my DB, as you can see. Common v4 chip, but with additional line (m-series have similar one at back, but 4 zones only).

**Reply by MuaazH** [reply]
Well I cloned this [T-Troll/AlienFX-SDK](https://github.com/T-Troll/AlienFX-SDK) and tried to use it for a simple console app, but I found out that this project is much cleaner, so I used it instead.
The docs were at the time saying use `SetColor(byte, byte, byte, byte)`, this method was removed or changed. It's all good now, the code is clean so I figured it out, and the docs are updated now.
> I have its mappings into my DB, as you can see. Common v4 chip, but with additional line (m-series have similar one at back, but 4 zones only).

Oh? I only see four devices. None of them is G7. If it's on some other file, then that's on me. 👍 
![Devices](https://user-images.githubusercontent.com/69297592/218669254-6baeaf30-8739-48bc-b707-f60b20b1066f.png)

**Reply by T-Troll** [reply]
I update it this morning, sorry, it was ancient version. So check new one.
`SetColor` now different - `SetColor(lightID, Afx_action c)`
Check `alienfx-sdk.h` for details - it's commented.

So it's common 0x550 chip. Check devices.csv - seems like you are 5500-compatible, but don't sure.
